services/graphql-server/src/graphql/definitions/email/campaign.js

```diff
@@ -1,11 +1,80 @@
 const gql = require('graphql-tag');
 
 module.exports = gql`
+  extend type Query {
+    emailCampaign(input: EmailCampaignQueryInput!): EmailCampaign
+      @findOne(model: "email.Campaign", using: { id: "_id" })
+    emailCampaigns(input: EmailCampaignQueryInput!): EmailCampaignConnection
+  }
 
-type EmailCampaignTestRecipient {
-  firstName: String
-  lastName: String
-  email: String
-}
+  input EmailCampaignQueryInput {
+    status: ModelStatus = active
+    id: ObjectID
+    productId: ObjectID
+    sort: EmailCampaignsSortInput = {}
+    pagination: PaginationInput = {}
+    beginning: CampaignBeginningInput = {}
+    ending: CampaignEndingInput = {}
+  }
+
+  input CampaignBeginningInput {
```

Comment:
The before/after beginning/ending inputs are sort of counter-intuitive. Do you need them? If not, I would just use a single date to indicate the start of your query, combined with sorting and/or limiting to get the result you want.

If you just need to find the latest campaign for example, you could just query
```gql
query($input: EmailCampaignsQueryInput!) {
  emailCampaigns(input: $input) {
    edges {
      node {
        id
        name
        scheduled
      }
    }
  }
}

// with input
{
  "input": {
    "productId": "1234mongoid",
    "scheduledAfter": 1234timestamp,
    "sort": {
      "field": "scheduled",
      "order": "desc"
    },
    "pagination": {
      "limit": 1
    }
  }
}
```

Reply:
That is fine I don't really need before/after. Is After just supported or is that something I implement like published-content.js?